src/mem/Allocator.h

```diff
@@ -29,22 +36,33 @@ class Allocator : public Interface
      */
     virtual size_t getStats(PoolStats &) = 0;
 
-    virtual PoolMeter const &getMeter() const = 0;
+    /// get current statistics tracked for this allocator
+    virtual PoolMeter const &getMeter() const { return meter; }
+    virtual PoolMeter &getMeter() { return meter; }
 
     /// provide (and reserve) memory suitable for storing one object
-    virtual void *alloc() = 0;
+    void *alloc() {
+        if (++count.allocs == FlushLimit)
+            flushCounters();
+        return allocate();
+    }
 
     /// return memory reserved by alloc()
-    virtual void freeOne(void *) = 0;
+    void freeOne(void *obj) {
+        assert(obj != nullptr);
+        (void) VALGRIND_CHECK_MEM_IS_ADDRESSABLE(obj, objectSize());
+        deallocate(obj);
+        ++count.freed;
+    }
 
     /// brief description of objects returned by alloc()
     virtual char const *objectType() const { return label; }
 
     /// the size (in bytes) of objects managed by this allocator
-    virtual size_t objectSize() const = 0;
+    size_t objectSize() const { return objectSize_; }
```

Comment:
Why not expose a constant obj_size data member? AFAICT, doing so will simplify/clarify the code (a long-term benefit) and significantly reduce the number of trivial code changes in this PR (a short term benefit).

I do not recommend doing that in _this_ PR (because it will add noise to an already noisy and complex changes), but we can also remove this method (after this PR adds a constant data member).

Reply:
Simply that this PR focus is for merging of the two classes. Polishing that is not directly necessary for that merge is out of scope.

Reply:
> this PR focus is for merging of the two classes. Polishing that is not directly necessary for that merge is out of scope.

This PR _renames_ the moved public data member and makes it _private_. Those two decisions/actions are clearly in this PR scope. Thus, the requested adjustments (of those actions) are also in scope. They also improve the overall PR outcome and are trivial.

Reply:
So are you standing by your claim "I do not recommend doing that in this PR"  ?? or actually insisting on the change?

Reply:
This change request discussed two changes:

* "expose a constant obj_size data member"; please implement that change in this PR
* "remove this method (after this PR adds a constant data member)"; I do not recommend doing that change in this PR

Reply:
Done in dc46e76be849276568edce5fd0f04e5279393990
I have done both changes because it makes no sense to expand the scope of this PR to polish the API, only to force an immediate followup PR to re-polish the polishing changes.